### PR TITLE
Resolve envoy service name from a different position

### LINKED ISSUE #6647: Envoy ALS mx-mesh should resolve service name from another position
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [ ] Question or discussion
- [ ] Bug
- [ ] Requirement
- [x] Feature or performance improvement
___
### Requirement or improvement
Since Istio injects ` service.istio.io/canonical-name` to the pod as the service name, OAP Envoy receiver should update `https://github.com/apache/skywalking/blob/master/oap-server/server-bootstrap/src/main/resources/metadata-service-mapping.yaml` to replace `LABELS.app` with this canonical label.

### PR DESCRIPTION
**Pull request by hanahmily**
* Using "service.istio.io/canonical-name" to replace "app" label to resolve service name
 * Improving FieldsHelper to support dot in labels

Fixed #6647

### CODE REVIEW

File: oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/mx/FieldsHelper.java [hunk 1]
```diff
@@ -82,7 +85,24 @@ public void init(final InputStream inputStream,
             final StringBuffer serviceNamePattern = new StringBuffer();
             while (m.find()) {
                 final String property = m.group("property");
-                flatBuffersFieldNames.add(Splitter.on('.').omitEmptyStrings().splitToList(property));
+                List<String> tokens = Splitter.on('.').omitEmptyStrings().splitToList(property);
```
Comment (kezhenxu94) [inline oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/mx/FieldsHelper.java:88]: Just did a quick Google search, replacing the `.` with this delimiter `\.(?=([^"]*"[^"]*")*[^"]*$)` can do the trick, but you can make the call whether to replace it or not

**Review by kezhenxu94 — DISMISSED** [review]
LGTM if tests pass

File: oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/mx/FieldsHelper.java (hunk 1 above)
Comment (wu-sheng) [reply oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/mx/FieldsHelper.java:88]: I prefer the current way. It is more clear.

**Comment by wu-sheng** [comment]
@hanahmily Code style issue? Please recheck.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**